addons/mail/static/src/utils/test_utils.js

```diff
@@ -169,47 +170,99 @@ function nextAnimationFrame() {
+ *
+ */
+async function setupModelDefinitions() {
+    const modelDefinitionsRegistry = registry.category('mail.model.definitions');
+    const modelNamesToFetch = modelDefinitionsRegistry.get('modelNamesToFetch');
+    const fieldsToInsertRegistry = modelDefinitionsRegistry.category('fieldsToInsert');
+
+    // fetch the model definitions
+    const formData = new FormData();
+    formData.append('csrf_token', core.csrf_token);
+    formData.append('model_names_to_fetch', JSON.stringify(modelNamesToFetch));
+    const response = await window.fetch('/mail/get_model_definitions', { body: formData, method: 'POST' });
+    if (response.status !== 200) {
+        throw new Error('Error while fetching required models');
+    }
+    modelDefinitions = new Map(Object.entries(await response.json()));
+
+    for (const [modelName, fields] of modelDefinitions) {
+         // insert fields present in the fieldsToInsert registry : if the field
+         // exist, update it's default value according to the one in the
+         // registry; If it does not exist, add it to the model definition.
+        const fieldNamesToFieldToInsert = fieldsToInsertRegistry.category(modelName).getEntries();
+        for (const [fname, fieldToInsert] of fieldNamesToFieldToInsert) {
+            if (fname in fields) {
+                fields[fname].default = fieldToInsert.default;
+            } else {
+                fields[fname] = fieldToInsert;
+            }
+        }
+        // apply default values for date like fields if none was passed
+        for(const fname in fields) {
+            const field = fields[fname];
+            if (['date', 'datetime'].includes(field.type) && !field.default) {
+                const defaultFieldValue =
+                    field.type === 'date'
+                    ? () => moment().format('YYYY-MM-DD')
+                    : () => moment.utc().format("YYYY-MM-DD HH:mm:ss")
+                field.default = defaultFieldValue;
+            }
+        }
+    }
+}
 
-    data.publicPartnerId = 4;
-    data['res.partner'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicPartnerId,
-    });
-    data.publicUserId = 3;
-    data['res.users'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicUserId,
-        name: "Public user",
-        partner_id: data.publicPartnerId,
-    });
+//------------------------------------------------------------------------------
+// Public: test lifecycle
+//------------------------------------------------------------------------------
+
+async function beforeEach(self) {
+    if (!modelDefinitions) {
+        await setupModelDefinitions();
+    }
+    const data = { ...TEST_USER_IDS };
+    const recordsToInsertRegistry = registry.category('mail.model.definitions').category('recordsToInsert');
+    for (const [modelName, fields] of modelDefinitions) {
+        let records = [];
```

Comment:
prefer const whenever possible (with conditional to assigned value if necessary)